src/cpp/src/block_manager.hpp

```diff
@@ -75,56 +78,81 @@ class KVCacheBlock {
 
 
 class Evictor {
```

Comment:
@popovaan please, review changes in this file as well.
The idea is to have different block tables for each decoder layer within a sequence